grakn-graph/src/main/java/ai/grakn/graph/internal/RelationImpl.java

```diff
@@ -58,20 +62,32 @@
     }
 
     /**
-     *
-     * @return All the castings this relation is connected with
+     * Sets the internal hash in order to perform a faster lookup
      */
-    Set<CastingImpl> getMappingCasting() {
-        Set<CastingImpl> castings = new HashSet<>();
-        getOutgoingNeighbours(Schema.EdgeLabel.CASTING).forEach(casting -> castings.add(((CastingImpl) casting)));
-        return castings;
+    void setHash(){
+        setUniqueProperty(Schema.ConceptProperty.INDEX, generateNewHash(type(), allRolePlayers()));
     }
 
     /**
-     * Sets the internal hash in order to perform a faster lookup
+     * Castings are retrieved from the perspective of the {@link Relation}
+     *
+     * @param roleTypes The role which the instances are playing
+     * @return The {@link Casting} which unify a {@link RoleType} and {@link Instance} with this {@link Relation}
      */
-    void setHash(){
-        setUniqueProperty(Schema.ConceptProperty.INDEX, generateNewHash(type(), allRolePlayers()));
+    Stream<Casting> castingsRelation(RoleType... roleTypes){
+        if(roleTypes.length == 0){
+            return getEdgesOfType(Direction.OUT, Schema.EdgeLabel.SHORTCUT).
+                    map(edge -> getGraknGraph().getElementFactory().buildRolePlayer(edge));
+        }
+
+        //Traversal is used so we can potentially optimise on the index
+        Set<Integer> roleTypesIds = Arrays.stream(roleTypes).map(r -> r.getTypeId().getValue()).collect(Collectors.toSet());
```

Comment:
No, this is used for internal testing + validation. Getting the roleplayers of the sub types will lead to extra processing for no reason. This method is not even on the public API.